docs/02-app/01-building-your-application/11-upgrading/05-from-create-react-app.mdx

```diff
@@ -0,0 +1,574 @@
+    </html>
+  )
+}
+```
+
+```jsx filename="app/layout.js" switcher
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <head>
+        <title>React App</title>
+        <meta name="description" content="Web site created..." />
+      </head>
+      <body>
+        <div id="root">{children}</div>
+      </body>
+    </html>
+  )
+}
+```
+
+6. Finally, Next.js can manage your last `<head>` tags with the
+   [Metadata API](/docs/app/building-your-application/optimizing/metadata). Move your final metadata
+   info into an exported
+   [`metadata` object](/docs/app/api-reference/functions/generate-metadata#metadata-object):
+
+```tsx filename="app/layout.tsx" switcher
+import type { Metadata } from 'next'
+
+export const metadata: Metadata = {
+  title: 'React App',
+  description: 'Web site created with Next.js.',
+}
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  return (
+    <html lang="en">
+      <body>
+        <div id="root">{children}</div>
+      </body>
+    </html>
+  )
+}
+```
+
+```jsx filename="app/layout.js" switcher
+export const metadata = {
+  title: 'React App',
+  description: 'Web site created with Next.js.',
+}
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <body>
+        <div id="root">{children}</div>
+      </body>
+    </html>
+  )
+}
+```
+
+With the above changes, you shifted from declaring everything in your `index.html` to using Next.js'
+convention-based approach built into the framework
+([Metadata API](/docs/app/building-your-application/optimizing/metadata)). This approach enables you
+to more easily improve your SEO and web shareability of your pages.
+
+### Step 5: Create the Entrypoint Page
+
+On Next.js you declare an entrypoint for your application by creating a `page.tsx` file. The
+closest equivalent of this file on CRA is your `src/index.tsx` file. In this step, you’ll set up the
+entrypoint of your application.
+
+1. **Create a `[[...slug]]` directory in your `app` directory.**
+
+Since in this guide we're aiming first to set up our Next.js as an SPA (Single Page Application),
+you need your page entrypoint to catch all possible routes of your application. For that, create a
+new `[[...slug]]` directory in your `app` directory.
+
+This directory is what is called an
+[optional catch-all route segment](/docs/app/building-your-application/routing/dynamic-routes#optional-catch-all-segments).
+Next.js uses a file-system based router where
+[directories are used to define routes](/docs/app/building-your-application/routing/defining-routes#creating-routes).
+This special directory will make sure that all routes of your application will be directed to its
+containing `page.tsx` file.
+
+2. **Create a new `page.tsx` file inside the `app/[[...slug]]` directory with the following content:**
+
+```tsx filename="app/[[...slug]]/page.tsx" switcher
+import '../../src/index.css'
```

Comment:
Would global CSS typically be imported in the Root Layout or Page? Does it matter?

Reply:
In this instance, they're equivalent since we're not using something like `app/about/page.tsx` anywhere else.